### PR TITLE
[4.1] RavenDB-11949 &  RavenDB-11441 &  RavenDB-12002 & RavenDB-12014 & RavenDB-12003

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-11949
http://issues.hibernatingrhinos.com/issue/RavenDB-11441
http://issues.hibernatingrhinos.com/issue/RavenDB-12002
http://issues.hibernatingrhinos.com/issue/RavenDB-12014
http://issues.hibernatingrhinos.com/issue/RavenDB-12003

### CODE REVIEW

**Review by ml054 — APPROVED** [review]
studio part looks good. :)

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTask.cs [hunk 1]
```diff
@@ -90,6 +90,9 @@ public async Task<IOperationResult> Execute(Action<IOperationProgress> onProgres
                         onProgress,
                         result);
 
+                    if (restoreSettings != null && _restoreConfiguration.SkipIndexesImport)
```
Comment (ayende) [inline src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTask.cs]: Shouldn't the logic here be reversed?

Comment (ayende) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTask.cs]: Okay, the code is correct, but _really_ confusing. 
Why do you need it here if this is empty all the time?

File: src/Raven.Client/Documents/Operations/Backups/RestoreBackupConfiguration.cs [hunk 2]
```diff
@@ -13,5 +13,7 @@ public class RestoreBackupConfiguration
         public string EncryptionKey { get; set; }
 
         public bool DisableOngoingTasks { get; set; }
+
+        public bool SkipIndexesImport { get; set; }
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/Backups/RestoreBackupConfiguration.cs]: rename to 'SkipIndexes'

Comment (grisha-kotler) [reply src/Raven.Client/Documents/Operations/Backups/RestoreBackupConfiguration.cs]: done

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTask.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTask.cs]: done

**Pull request merged**